test/test_distributions.py

```diff
@@ -705,13 +739,22 @@ def test_beta_shape_tensor_params(self):
         self.assertRaises(ValueError, dist.log_prob, self.tensor_sample_2)
 
     def test_categorical_shape(self):
-        categorical = Categorical(torch.Tensor([[0.6, 0.3], [0.6, 0.3], [0.6, 0.3]]))
-        self.assertEqual(categorical._batch_shape, torch.Size((3,)))
-        self.assertEqual(categorical._event_shape, torch.Size(()))
-        self.assertEqual(categorical.sample().size(), torch.Size((3,)))
-        self.assertEqual(categorical.sample((3, 2)).size(), torch.Size((3, 2, 3,)))
-        self.assertRaises(ValueError, categorical.log_prob, self.tensor_sample_1)
-        self.assertEqual(categorical.log_prob(self.tensor_sample_2).size(), torch.Size((3, 2, 3)))
+        dist = Categorical(torch.Tensor([[0.6, 0.3], [0.6, 0.3], [0.6, 0.3]]))
+        self.assertEqual(dist._batch_shape, torch.Size((3,)))
+        self.assertEqual(dist._event_shape, torch.Size(()))
+        self.assertEqual(dist.sample().size(), torch.Size((3,)))
+        self.assertEqual(dist.sample((3, 2)).size(), torch.Size((3, 2, 3,)))
+        self.assertRaises(ValueError, dist.log_prob, self.tensor_sample_1)
+        self.assertEqual(dist.log_prob(self.tensor_sample_2).size(), torch.Size((3, 2, 3)))
+
+    def test_one_hot_categorical_shape(self):
+        dist = OneHotCategorical(torch.Tensor([[0.6, 0.3], [0.6, 0.3], [0.6, 0.3]]))
+        self.assertEqual(dist._batch_shape, torch.Size((3,)))
+        self.assertEqual(dist._event_shape, torch.Size((2,)))
+        self.assertEqual(dist.sample().size(), torch.Size((3, 2)))
+        self.assertEqual(dist.sample((3, 2)).size(), torch.Size((3, 2, 3, 2)))
+        self.assertRaises(ValueError, dist.log_prob, self.tensor_sample_2)
```

Comment:
Could you also validate on the second sample:
`self.assertEqual(dist.log_prob(self.tensor_sample_1).size(), torch.Size((3,)))`

This was one of our test cases in https://docs.google.com/a/uber.com/spreadsheets/d/1V26WwfPoj7o0f-B9jMNAlxzogE_N83Sih04ZjbeymgU/edit?usp=sharing?